pycbc/io/live.py

```diff
@@ -388,18 +388,53 @@ def upload(self, fname, gracedb_server=None, testing=True,
                 self.gracedb.write_label(gid, 'INJ')
                 logging.info("Tagging event %s as an injection", gid)
 
-            # add info for tracking code version
-            gracedb_tag_with_version(self.gracedb, gid)
-
-            extra_strings = [] if extra_strings is None else extra_strings
-            for text in extra_strings:
-                self.gracedb.write_log(gid, text, tag_name=['analyst_comments'])
         except Exception as exc:
-            logging.error('Something failed during the upload/annotation of '
+            logging.error('Something failed during the upload of '
                           'event %s on GraceDB. The event may not have been '
                           'uploaded!', fname)
             logging.error(str(exc))
 
+        # Upload em_bright properties JSON
+        if self.hasmassgap is not None:
```

Comment:
You should also check that `gid is not None` here, and in the similar blocks below, as the upload could have failed, but we still want to plot things in that case.

Reply:
And you may want to correspondingly remove the "Don't try to do anything else!" block further down.

Reply:
Hmm, OK, this aspect (what to do in the absence of gid) should probably be handled by a general upload method, but that is a bit more of a rewrite.  ATM if gid is `None` then the uploads should fail and the error messages will be logged, correct?

Reply:
True, assuming `write_log()` does not do anything catastrophically unpleasant when given None… Which is probably the case.

Reply:
I think anything short of a segmentation fault should be covered